.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,61 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - '[rv][0-9].[0-9].[0-9]'
+      - '[rv][0-9].[0-9].[0-9]rc[0-9]'
+    types: [ labeled ]
+
+jobs:
+  build-container:
+    if: ${{ github.event.label.name == 'gpuci' }}
+    uses: ./.github/workflows/_build_container.yml
+
+  gpu-test:
+    needs: build-container
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v4
```

Comment:
Can also be dropped (we run everything inside the container where the source code already exists - so no need to checkout again)